### DISCUSSION TITLE
v7.3.0-preview.2 Release of PowerShell

### DISCUSSION THREAD

**Discussion by TravisEz13** [PowerShell/PowerShell#16919] [Q&A]
### Engine Updates and Fixes

- Fix the `clean` block for generated proxy function (#16827)
- Add support to allow invoking method with generic type arguments (#12412 and #16822) (Thanks @vexx32!)
- Report error when PowerShell built-in modules are missing (#16628)

### General Cmdlet Updates and Fixes

- Prevent command completion if the word to complete is a single dash (#16781) (Thanks @ayousuf23!)
- Use `FindFirstFileW` instead of `FindFirstFileExW` to correctly handle Unicode file names on FAT32 (#16840) (Thanks @iSazonov!)
- Add completion for loop labels after Break/Continue (#16438) (Thanks @MartinGC94!)
- Support OpenSSH options for `PSRP` over SSH commands (#12802) (Thanks @BrannenGH!)
- Adds a `.ResolvedTarget` Property to `File-System` Items to Reflect a Symlink's Target as `FileSystemInfo` (#16490) (Thanks @hammy3502!)
- Use `NotifyEndApplication` to re-enable VT mode (#16612)
- Add new parameter to `Start-Sleep`: `[-Duration] <timespan>` (#16185) (Thanks @IISResetMe!)
- Add lock and null check to remoting internals (#16542) (#16683) (Thanks @SergeyZalyadeev!)
- Make `Measure-Object` ignore missing properties unless running in strict mode (#16589) (Thanks @KiwiThePoodle!)
- Add `-StrictMode` to `Invoke-Command` to allow specifying strict mode when invoking command locally (#16545) (Thanks @Thomas-Yu!)
- Fix `$PSNativeCommandArgPassing` = `Windows` to handle empty args correctly (#16639)
- Reduce the amount of startup banner text (#16516) (Thanks @rkeithhill!)
- Add `exec` cmdlet for bash compatibility (#16462)
- Add AMSI method invocation logging as experimental feature (#16496)
- Fix web cmdlets so that an empty `Get` does not include a `content-length` header (#16587)
- Update `HelpInfoUri` for 7.3 release (#16646)
- Fix parsing `SemanticVersion` build label from version string (#16608)
- Fix `ForEach-Object -Parallel` when passing in script block variable (#16564)

### Code Cleanup

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@eltociear, @iSazonov, @xtqqczze</p>

</summary>

<ul>
<li>Fix typo in PowerShellExecutionHelper.cs (#16776) (Thanks @eltociear!)</li>
<li>Use more efficient platform detection API (#16760) (Thanks @iSazonov!)</li>
<li>Seal <code>ClientRemotePowerShell</code> (#15802) (Thanks @xtqqczze!)</li>
<li>Fix the DSC overview URL in a markdown file and some small cleanup changes (#16629)</li>
</ul>

</details>

### Tools

- Fix automation to update experimental JSON files in GitHub action (#16837)

### Tests

- Update `markdownlint` to the latest version (#16825)
- Bump the package `path-parse` from `1.0.6` to `1.0.7` (#16820)
- Remove assert that is incorrect and affecting our tests (#16588)

### Build and Packaging Improvements

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@dahlia</p>

</summary>

<ul>
<li>Update NuGet Testing to not re-install dotnet,
when not needed and dynamically determine the DOTNET_ROOT (Internal 19268, 19269, 19272, 19273, and 19274)</li>
<li>Remove SkipExperimentalFeatureGeneration when building alpine (Internal 19248)</li>
<li>Revert .NET 7 changes, Update to the latest .NET 6 and Update WXS file due to blocking issue in .NET 7 Preview 1 </li>
<li>Install and Find AzCopy</li>
<li>Use Start-PSBootStrap for installing .NET during nuget packaging</li>
<li>Fix pool syntax for deployments (Internal 19189)</li>
<li>Bump NJsonSchema from 10.5.2 to 10.6.9 (#16888)</li>
<li>Update projects and scripts to use .NET 7 preview 1 pre-release builds (#16856)</li>
<li>Add warning messages when package precheck fails (#16867)</li>
<li>Refactor Global Tool packaging to include SBOM generation (#16860)</li>
<li>Update to use <code>windows-latest</code> as the build agent image (#16831)</li>
<li>Ensure alpine and arm SKUs have <code>powershell.config.json</code> file with experimental features enabled (#16823)</li>
<li>Update experimental feature json files (#16838) (Thanks @github-actions[bot]!)</li>
<li>Remove WiX install (#16834)</li>
<li>Add experimental json update automation (#16833)</li>
<li>Update .NET SDK to 6.0.101 and fix <code>Microsoft.PowerShell.GlobalTool.Shim.csproj</code> (#16821)</li>
<li>Add SBOM manifest to nuget packages (#16711)</li>
<li>Improve logic for updating .NET in CI (#16808)</li>
<li>Add Linux package dependencies for packaging (#16807)</li>
<li>Switch to our custom images for build and release (#16801)</li>
<li>Remove all references to <code>cmake</code> for the builds in this repo (#16578)</li>
<li>Fix build for new <code>InvokeCommand</code> attributes (#16800)</li>
<li>Let macOS installer run without Rosetta on Apple Silicon (#16742) (Thanks @dahlia!)</li>
<li>Update the expect .NET SDK quality to GA for installing dotnet (#16784)</li>
<li>Change nuget release yaml to use <code>UseDotNet</code> task (#16701)</li>
<li>Bump Microsoft.ApplicationInsights from 2.19.0 to 2.20.0 (#16642)</li>
<li>Register NuGet source when generating <code>CGManifest</code> (#16570)</li>
<li>Update Images used for release (#16580)</li>
<li>Update SBOM generation (#16641)</li>
<li>Bring changes from 7.3.0-preview.1 (#16640)</li>
<li>Update the <code>vmImage</code> and PowerShell root directory for macOS builds (#16611)</li>
<li>Update macOS build image and root folder for build (#16609)</li>
<li>Disabled Yarn cache in markdown.yml (#16599)</li>
<li>Update cgmanifest (#16600)</li>
<li>Fix broken links in markdown (#16598)</li>
</ul>

</details>

### Documentation and Help Content

- Add newly joined members to their respective Working Groups (#16849)
- Update Engine Working Group members (#16780)
- Replace the broken link about pull request (#16771)
- Update change log to remove a broken URL (#16735)
- Updated `README.md` and `metadata.json` for `v7.3.0-preview.1` release (#16627)
- Updating changelog for `7.2.1` (#16616)
- Updated `README.md` and `metadata.json` for `7.2.1` release (#16586)

[7.3.0-preview.2]: https://github.com/PowerShell/PowerShell/compare/v7.3.0-preview.1...v7.3.0-preview.2

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - 9622CC0F25B9320E8AA0C4EC1A716174386C60788FF58EB20C52627CB732A69B
- powershell-7.3.0-preview.2-linux-alpine-x64.tar.gz
  - A2AB96E383C4B6A5F689610AA3CFAF1428D5FBEFC88C4FEA1AED6C4061CE5672
- powershell-7.3.0-preview.2-linux-arm32.tar.gz
  - 37C74FE1253A4DEE44BC06647726FCB74A882FD3C13B026C3E56C85A203E7133
- powershell-7.3.0-preview.2-linux-arm64.tar.gz
  - 1D54441B686E52B209F317CC33875374E5E92D0E547B1BCC38F3240444801A98
- powershell-7.3.0-preview.2-linux-x64.tar.gz
  - 03A97A27FA076A948C6E56A59B431ACBD04F4FF930A4027DD877C10EBDE11DA8
- powershell-7.3.0-preview.2-linux-x64-fxdependent.tar.gz
  - 2371760BCEF5D4ED8F3C738FD3FBFCDD355ED74E89DE3D7B35CA646B9A550A73
- powershell-7.3.0-preview.2-osx-arm64.pkg
  - F1DD9572CE7E696764D8192250E73ECD6A064C0A95174373C6232729458CD62D
- powershell-7.3.0-preview.2-osx-arm64.tar.gz
  - 67B35C3643C7218645FA0A2BB4627FC1A3006E09C09EBEC85C99C78C1F9AEC46
- powershell-7.3.0-preview.2-osx-x64.pkg
  - A582622CA7BBB818B712394EB060D4056D26AEB082831E27CD647FCED9A64C01
- powershell-7.3.0-preview.2-osx-x64.tar.gz
  - 49F6DFBFC0F03A120B08106B311FDCCE5075E53482597CFD91DFC8106CEF91BA
- PowerShell-7.3.0-preview.2-win-arm32.zip
  - 81B125397CCAB0D9FDECE7F92D26213B7C7D92ED14011B06BDB197CF158692EF
- PowerShell-7.3.0-preview.2-win-arm64.zip
  - 5034CD1A84AA3365DC829468DB2A62B030AF738B40312499A2D1586F6206964A
- PowerShell-7.3.0-preview.2-win-fxdependent.zip
  - 5F0956B7C1214DF73D23022BAB050D4B7F145B1DCF64EAF6E2CEF1FC95ED1CB5
- PowerShell-7.3.0-preview.2-win-fxdependentWinDesktop.zip
  - 791D32319158C6DF96E5C4429C0B6245DF975C068A29E9DEAD0D4C36A0B1D90D
- PowerShell-7.3.0-preview.2-win-x64.msi
  - 8C2392E4A7264AD7AC0655C46AB35545A7744110D192ACA8164767A9D3CD4CF8
- PowerShell-7.3.0-preview.2-win-x64.zip
  - C62A26B8322330D27DCCEE007924A9429DA0F2B33B5EAD2656CBDF95FFBD2268
- PowerShell-7.3.0-preview.2-win-x86.msi
  - E85C9E08CB9B7332002F3766BECC8062374EA7D32053F74A53898E00667F4E8C
- PowerShell-7.3.0-preview.2-win-x86.zip
  - 754F93A62696A12BA49A6DF4F266BAEBE34E88C5BB305E54EA1C0019B0511156
- powershell-preview_7.3.0-preview.2-1.deb_amd64.deb
  - 308CDD116AC1215321CB811254EBF682A228DA24B811538D4336F837A283A490
- powershell-preview-7.3.0_preview.2-1.rh.x86_64.rpm
  - 2E07548A2757A7026EA8221C595E3AEB8576C12DDAE170BA816719A34212DB13

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.3.0-preview.2'>v7.3.0-preview.2 Release of PowerShell</a>.</em>

**Comment by dahlia** [comment]
You can install this preview version on your Mac with Homebrew:

```bash
brew install homebrew/cask-versions/powershell-preview
```

Also note that since this release PowerShell installer for Mac does not require Rosetta 2 at all on Apple Silicon Macs (i.e., M1, M1 Pro, M1 Max).

**Comment by kasini3000** [comment]
Congratulations! 
Currently only rhel package, 
Please release debian and ubuntu packages. i want to try
http://packages.microsoft.com/ubuntu/20.04/prod/pool/main/p/powershell-preview/

**Reply by TravisEz13** [reply]
Sorry, I'm only looking at this and preview.3 is out.  I don't know how this happened.
